routingpy/routers/vroom.py

```diff
@@ -0,0 +1,173 @@
+        **client_kwargs
+    ):
+        """
+        Initializes a Vroom client.
+
+        :param base_url: The base URL for the request.
+        :type base_url: str
+
+        :param user_agent: User Agent to be used when requesting.
+            Default :attr:`routingpy.routers.options.default_user_agent`.
+        :type user_agent: str
+
+        :param timeout: Combined connect and read timeout for HTTP requests, in
+            seconds. Specify ``None`` for no timeout. Default :attr:`routingpy.routers.options.default_timeout`.
+        :type timeout: int or None
+
+        :param retry_timeout: Timeout across multiple retriable requests, in
+            seconds.  Default :attr:`routingpy.routers.options.default_retry_timeout`.
+        :type retry_timeout: int
+
+        :param retry_over_query_limit: If True, client will not raise an exception
+            on HTTP 429, but instead jitter a sleeping timer to pause between
+            requests until HTTP 200 or retry_timeout is reached.
+            Default :attr:`routingpy.routers.options.default_retry_over_query_limit`.
+        :type retry_over_query_limit: bool
+
+        :param skip_api_error: Continue with batch processing if a :class:`routingpy.exceptions.RouterApiError` is
+            encountered (e.g. no route found). If False, processing will discontinue and raise an error.
+            Default :attr:`routingpy.routers.options.default_skip_api_error`.
+        :type skip_api_error: bool
+
+        :param client: A client class for request handling. Needs to be derived from :class:`routingpy.base.BaseClient`
+        :type client: abc.ABCMeta
+
+        :param **client_kwargs: Additional arguments passed to the client, such as headers or proxies.
+        :type **client_kwargs: dict
+        """
+
+        self.client = client(
+            base_url,
+            user_agent,
+            timeout,
+            retry_timeout,
+            retry_over_query_limit,
+            skip_api_error,
+            **client_kwargs
+        )
+
+    def optimization(
+        self, vehicles, jobs=None, shipments=None, matrices=None, geometry=False, dry_run=None
+    ):
+        """
+        Optimize a fleet of vehicles on a number of jobs.
+        For more information, visit https://github.com/VROOM-Project/vroom/blob/master/docs/API.md.
+        Example:
+            >>> from routingpy.routers import Vroom
+            >>> from routingpy.optimization import Job, Vehicle
+            >>> vehicles = [Vehicle(0, start=[7.414210118164926, 43.72810792282373], end=[7.414210118164926, 43.72810792282373])]
+            >>> jobs = [Job(0, location=[7.423805608013283, 43.739198887075084]), Job(1, location=[7.414257185161816, 43.73378896481369])]
+            >>> v = Vroom("http://localhost:3000")
+            >>> o = v.optimization(jobs, vehicles, geometry=True)
+        :param vehicles: array of vehicle objects describing the available vehicles.
+        :type vehicles: list of Vehicle
```

Comment:
for vroom proper type hinting of all these custom objects is the most important thing, otherwise IDE's intellisense won't pick up the object's definitions